docs/ruby_on_rails/docker_with_ror.md

```diff
@@ -0,0 +1,166 @@
+
+## 3. Cấu Hình Docker
+### 3.1. Tạo Dockerfile
+#### Mục Đích
+
+Giải thích mục đích của Dockerfile trong việc xây dựng Docker image.
+
+#### Nội Dung
+
+- Cài đặt Ruby và các dependencies.
+- Sao chép mã nguồn vào container.
+- Thiết lập lệnh khởi động container.
+
+- File cấu hình:
+
+```DockerFile
+# Sử dụng image của Ruby
+FROM ruby:3.1
+
+ENV APP_ROOT /app
+# Cài đặt các gói phụ thuộc
+RUN apt-get update -qq && apt-get install -y \
+    build-essential \
+    libpq-dev \
+    nodejs \
+    yarn
+
+# Thiết lập thư mục làm việc
+WORKDIR /app
+
+# Sao chép file Gemfile và Gemfile.lock vào thư mục làm việc
+COPY ./Gemfile $APP_ROOT
+COPY ./Gemfile.lock $APP_ROOT
+
+# Cài đặt các gem được liệt kê trong Gemfile
+RUN bundle install
+
+# Sao chép toàn bộ mã nguồn ứng dụng vào thư mục làm việc
+COPY . $APP_ROOT
+
+# Timezone (nếu cần)
+RUN ln -sf  /usr/share/zoneinfo/Asia/Ho_CHi_Minh /etc/localtime
+```
+
+### 3.2. Tạo docker-compose.yml
+#### Mục Đích
+
+Giải thích vai trò của docker-compose.yml trong việc quản lý các dịch vụ.
+
+#### Nội Dung
+- Cấu hình dịch vụ cho Ruby on Rails.
+- Cấu hình dịch vụ cho cơ sở dữ liệu (PostgreSQL, MySQL, etc.).
+- Cấu hình mạng và volume.
+
+- File cấu hình: [docker-compose](/../static/docker_with_ror/docker-compose.yml)
+
+### 3.3. Tạo .dockerignore
+#### Mục Đích
+Giải thích vai trò của .dockerignore trong việc loại bỏ các tập tin không cần thiết khi build Docker image.
+
+#### Nội Dung
+
+Các mục nên được loại bỏ như thư mục node_modules, log, tmp, etc.
+
+- File cấu hình:
+
+```.dockerignore
+.git
+log/*
+tmp/*
+!.gitkeep
+.bundle
+vendor/bundle
+node_modules
+```
+
+### 3.4 Cấu hình database, mailhog, sideki
+
+Các file cấu hình như database.yml, mailhog, sidekiq.yml vẫn được cấu hình như bình thường, có thể tham khảo                                                                                                                                                                                                       
+
+- Database.yml: [database.yml](/../static/docker_with_ror/database.yml)
+- Sidekiq.yml: [sidekiq.yml](/../static/docker_with_ror/sidekiq.yml)   
+- Mailhog:
+```
+#config/environments/development.rb
+
+config.action_mailer.delivery_method = :smtp
+config.action_mailer.smtp_settings = {
+  address: ENV['MAILHOG_HOST'],
+  port: 1025
+}
+```                                                                                                                                                                                                                                                                                                                                      
+## 4. Một số CLI rails hay được sử dụng với Docker
+
+### 4.1. docker build, up, down
+
+```
+docker-compose build
+docker-compose up
+docker-compose down
+```
+
+### 4.2. docker start, restart, stop container
+
+```
+docker-compose start {container_id}
+docker-compose restart {container_id}
+docker-compose stop {container_id}
```

Comment:
`docker-compose` cli thì nó lại ko phải container id a ạ =)) nó tương tác qua service name, nãy a viết service id cũng được r. Nó khác với `docker` cli

Reply:
fixed in #14